### PR TITLE
allow users to hide columns programmatically

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated components in Deploy Preview](https://deploy-preview-517--neo-react-library-storybook.netlify.app/?path=/story/components-table--advanced-filtering-and-sorting)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [ ] ~tagged `@avaya-dux/dux-design` if any visual changes have occurred~
- [x] tagged `@avaya-dux/dux-devs`

Allow users to programmatically set hidden columns.

`@avaya-dux/dux-devs`

- I have extended `Column` and named it `NeoColumn` in order to allow this to work. I have updated all of our code to reflect this.
- The only real change is in `Table.tsx`.
- The linked story hides the "Long Text" column by default.
- The filter columns dialog still works as expected.

## Summary by CodeRabbit

- **New Features**
	- Introduced a `hiddenColumns` property to the Table component for improved control over column visibility.
	- Updated the Advanced Filtering and Sorting story to hide the "Long Text" column by default, enhancing user experience.
- **Style**
	- Added new CSS for extra spacing around sections to improve layout and readability in the Table component's story.
- **Documentation**
	- Expanded documentation to clarify how to hide columns using the `show: false` property.
- **Tests**
	- Added a new test case to verify the functionality of predefined hidden columns within the Table component.
- **Chores**
	- Updated the version of the `@avaya/neo-react` package from `1.2.6` to `1.2.7`.

### CODE REVIEW

File: src/components/Table/Table.stories.tsx [hunk 1]
```diff
@@ -405,18 +407,25 @@
 			Header: "Long Text",
 			accessor: "longText",
 			disableFilters: true,
+			show: false,
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Table/Table.stories.tsx:411]: updated TS to properly use `show`

File: src/components/Table/types/TableTypes.ts [hunk 2]
```diff
@@ -11,6 +11,10 @@ import type {
 // biome-ignore lint/suspicious/noExplicitAny: We require maximum flexibility here
 type AnyRecord = Record<string, any>;
 
+export type NeoColumn<T extends AnyRecord> = Column<T> & {
```
Comment (yangAtSpoken) [inline src/components/Table/types/TableTypes.ts:14]: Do not like this name.  How about VisibleColumn?

**Review by yangAtSpoken — COMMENTED** [review]
I expect seeing the columns prop of Table takes a type of VisibleColumn[], but not seeing it.

**Review by yangAtSpoken — CHANGES_REQUESTED** [review]
[This](https://github.com/TanStack/table/discussions/2421) looks like the proper way of extending Column type in Typescript.

**Comment by yangAtSpoken** [comment]
> > I expect seeing the columns prop of Table takes a type of VisibleColumn[], but not seeing it.
> 
> I'm not sure what you're asking for, but a user of the Table simply adds `show: false` to their column. Example in the storybook can be [seen here](https://github.com/avaya-dux/neo-react-library/pull/517/files#diff-27f14c5125072726601ca9775063ccab3f6d656cb5802c0c556d3f25071767e2R411).
> 
> If you mean in our implementation, the `Table.tsx` [now uses](https://github.com/avaya-dux/neo-react-library/pull/517/files#diff-94371dfd292919a830ac1ff2b4460ad28f30a81fa2b76acd87c0538a78a63a50R155) the built in `hiddenColumns` to hide columns.
> 
> Let me know if this doesn't answer your question.

Since NeoColumn extends Column, type of the columns prop  to the table should be changed to readonly NeoColumn<T>[].
Screenshot 2024-09-04 at 12 15 09 PM

**Comment by yangAtSpoken** [comment]
> > [This](https://github.com/TanStack/table/discussions/2421) looks like the proper way of extending Column type in Typescript.
> 
> I'm not sure that adding the extra three types (`UseFiltersColumnOptions`, `UseGroupByColumnOptions`, `UseSortByColumnOptions`) is necessary, but I can't think of why that would hurt anything, so I've updated to include those types in the `NeoColumn` definition.

The important part the line of collapsed?: boolean.  Instead of collapsed, show?: boolean should be there.  Since it is not easy to replace Column throughout react-table, adding the extra prop to Column directly is the way they recommended.

**Pull request merged**